webhook/resourcesemantics/validation/controller.go

```diff
@@ -43,12 +43,46 @@ func NewAdmissionControllerWithConfig(
 	handlers map[schema.GroupVersionKind]resourcesemantics.GenericCRD,
 	wc func(context.Context) context.Context,
 	disallowUnknownFields bool,
-	callbacks map[schema.GroupVersionKind]Callback,
+	callbacks ...map[schema.GroupVersionKind]Callback,
 ) *controller.Impl {
+
+	// This not ideal, we are using a variadic argument to effectively make callbacks optional
+	// This allows this addition to be non-breaking to consumers of /pkg
+	// TODO: once all sub-repos have adopted this, we might move this back to a traditional param.
+	var unwrappedCallbacks map[schema.GroupVersionKind]Callback
+	switch len(callbacks) {
+	case 0:
+		unwrappedCallbacks = map[schema.GroupVersionKind]Callback{}
+	case 1:
+		unwrappedCallbacks = callbacks[0]
+	default:
+		panic("NewAdmissionController may not be called with multiple callback maps")
+	}
+
+	opts := []OptionFunc{
+		WithPath(path),
+		WithTypes(handlers),
+		WithWrapContext(wc),
+		WithCallbacks(unwrappedCallbacks),
+	}
```

Comment:
@pierDipi I'm afraid this change would cause some other breaking changes to the current calls done to these functions. I don't want to deal with other battle changing the arguments to these controllers. I'd say this could be changed in a separated PR.

Reply:
Can you expand on why is that?

callbacks parameter is a single map currently:
```
callbacks map[schema.GroupVersionKind]Callback,
```
you're making it a varargs:
```
callbacks ...map[schema.GroupVersionKind]Callback,
```
to then add the hack and make it back a single map:
```
	// This not ideal, we are using a variadic argument to effectively make callbacks optional
	// This allows this addition to be non-breaking to consumers of /pkg
	// TODO: once all sub-repos have adopted this, we might move this back to a traditional param.
	var unwrappedCallbacks map[schema.GroupVersionKind]Callback
	switch len(callbacks) {
	case 0:
		unwrappedCallbacks = map[schema.GroupVersionKind]Callback{}
	case 1:
		unwrappedCallbacks = callbacks[0]
	default:
		panic("NewAdmissionController may not be called with multiple callback maps")
	}
```

therefore, I don't get why we need all of that in this case?

Reply:
Okay I see what you meant now. Github was failing to me, I changed `callbacks ...map[schema.GroupVersionKind]Callback,`  unintentional but I got confused with your comment. I can made it back to a single map.

Reply:
Fixed!